src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -166,9 +166,9 @@ - (instancetype)initWithNodeID:(uint64_t)nodeID deviceController:(MTRDeviceContr
     return self;
 }
 
-+ (instancetype)deviceWithNodeID:(uint64_t)nodeID deviceController:(MTRDeviceController *)deviceController
++ (instancetype)deviceWithNodeID:(NSNumber *)nodeID controller:(MTRDeviceController *)controller
```

Comment:
Not changed in this PR, but it looks to me like the return type should be `MTRDevice *` instead of `instancetype`? If you were to call this method on a sub-class of `MTRDevice`, would you be guaranteed to get an instance of that sub-class?

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/23384 to sort this out.